stage0/run.go

```diff
@@ -491,6 +492,19 @@ func Run(cfg RunConfig, dir string, dataDir string) {
 		args = append(args, "--local-config="+cfg.LocalConfig)
 	}
 
+	s1v, err := getStage1InterfaceVersion(dir)
+	if err != nil {
+		log.FatalE("error determining stage1 interface version", err)
+	}
+
+	if cfg.Hostname != "" {
+		if s1v > 1 {
```

Comment:
Or maybe make s1v an object instead of an integer, so you could add functions to it like `func (s1v Stage1InterfaceVersion) SupportsHostname() bool {}`

That would be an idea for a separate PR. ;P